tests/suites/test_suite_psa_crypto.function

```diff
@@ -6676,6 +6786,123 @@ exit:
+    size_t signature_size = signature_size_arg;
+    psa_status_t actual_status;
+    psa_status_t expected_start_status = expected_start_status_arg;
+    psa_status_t expected_complete_status = expected_complete_status_arg;
+    unsigned char *signature = NULL;
+    size_t signature_length = 0xdeadbeef;
+    size_t num_ops = 0;
+    size_t num_ops_prior = 0;
+    size_t num_completes = 0;
+    size_t min_completes = 0;
+    size_t max_completes = 0;
+
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+    psa_sign_hash_interruptible_operation_t operation =
+                                psa_sign_hash_interruptible_operation_init( );
+
+    ASSERT_ALLOC( signature, signature_size );
+
+    PSA_ASSERT( psa_crypto_init( ) );
+
+    psa_set_key_usage_flags( &attributes, PSA_KEY_USAGE_SIGN_HASH );
+    psa_set_key_algorithm( &attributes, alg );
+    psa_set_key_type( &attributes, key_type );
+
+    PSA_ASSERT( psa_import_key( &attributes, key_data->x, key_data->len,
+                                &key ) );
+
+    psa_interruptible_set_max_ops( max_ops );
+
+    if( max_ops == PSA_INTERRUPTIBLE_MAX_OPS_UNLIMITED )
+    {
+        min_completes = 1;
+        max_completes = 1;
+    }
+    else
+    {
+        /* Unfortunate, but failure cases tend to fail on the first op. */
+        if( expected_complete_status == PSA_SUCCESS )
+            min_completes = 2;
+        else
+            min_completes = 1;
+
+        max_completes = PSA_INTERRUPTIBLE_MAX_OPS_UNLIMITED;
+    }
+
+    num_ops_prior = psa_sign_hash_get_num_ops( &operation );
+    TEST_ASSERT( num_ops_prior == 0 );
+
+    /* Start performing the signature. */
+    actual_status = psa_sign_hash_start( &operation, key, alg,
+                                         input_data->x, input_data->len );
+
+    TEST_EQUAL( actual_status, expected_start_status );
+
+    num_ops_prior = psa_sign_hash_get_num_ops( &operation );
+    TEST_ASSERT( num_ops_prior == 0 );
+
+    /* Continue performing the signature until complete. */
+    do
+    {
+        actual_status = psa_sign_hash_complete( &operation, signature,
+                                                signature_size,
+                                                &signature_length );
+
+        num_completes++;
+
+        if( actual_status == PSA_SUCCESS ||
+            actual_status == PSA_OPERATION_INCOMPLETE )
+        {
+            num_ops = psa_sign_hash_get_num_ops( &operation );
+            TEST_ASSERT( num_ops > num_ops_prior );
+
+            num_ops_prior = num_ops;
+        }
+    }
+    while( actual_status == PSA_OPERATION_INCOMPLETE );
+
+    /* If the psa_sign_hash_start() failed, psa_sign_hash_complete()
+     * should also fail with bad state. */
+    if( expected_start_status != PSA_SUCCESS )
+        TEST_EQUAL( actual_status, PSA_ERROR_BAD_STATE );
+    else if( actual_status != PSA_OPERATION_INCOMPLETE )
+        TEST_EQUAL( actual_status, expected_complete_status );
```

Comment:
Why not unconditionally `TEST_EQUAL( actual_status, expected_complete_status );` here? The test code is validating the test data and I don't think that's necessary.